index.js

```diff
@@ -1,26 +1,156 @@
 const core = require('@actions/core');
 const github = require('@actions/github');
+const list = require('markdown-list');
+
+export function formatChType(type) {
+  const featureRegex = /(feat+)/;
+  if (!type) {
+    return 'other';
+  }
+
+  if (type.match(featureRegex)) {
+    return 'feature';
+  }
+
+  return type.replace(/[^a-zA-Z ]/g, '');
+}
+
+export function formatCommits(commits, chStoryUrl) {
+  const {
+    payload: {
+      repository: { url: repoUrl },
+    },
+  } = github.context;
+
+  return commits.reduce((acc, com) => {
+    const {
+      commit: { message },
+      sha: origSha,
+    } = com;
+    const regex = /(?<chType>\(\w*\))?(\s)?(?<prMsg>\w*\W*.+?)(\s+)?(\[)?(ch)?(?<chId>\d+)?(\])?\s\(#(?<prNumber>\d+)\)/;
+    const matches = message.match(regex);
+    const chType = formatChType(matches && matches.groups.chType);
+    const prMsg = (matches && matches.groups.prMsg) || message;
+    const prLink =
+      (matches &&
+        matches.groups.prNumber &&
+        `[#${matches.groups.prNumber}](${repoUrl}/pull/${matches.groups.prNumber})`) ||
+      null;
+    const chLink =
+      (matches &&
+        matches.groups.chId &&
+        `[ch${matches.groups.chId}](${chStoryUrl}/${matches.groups.chId})`) ||
+      null;
+    const sha = origSha.substring(0, 6);
+
+    const formattedCommit = { chLink, prMsg, prLink, sha };
+
+    return Object.assign(acc, {
+      [chType]: [...((acc[chType] && acc[chType]) || []), formattedCommit],
```

Comment:
Its not ;) On first run, `acc[chType]` does not exist.

Reply:
Hmm then something isn't right here. Given that this is a synchronous action let's say `acc[chType]` is `undefined` this code would read as `[...((undefined && undefined) || [])]`.  If something is happening between the `&&` line it would be very fragile code.

Reply:
Ohh! Woops! Good catch